tests/integration/src/test/java/org/apache/pulsar/tests/integration/auth/token/PulsarTokenAuthenticationBaseSuite.java

```diff
@@ -0,0 +1,178 @@
+    public void setupCluster() throws Exception {
+        // Before starting the cluster, generate the secret key and the token
+        // Use Zk container to have 1 container available before starting the cluster
+        try (ZKContainer zkContainer = new ZKContainer<>("cli-setup")) {
+            zkContainer
+                    .withEnv("zkServers", ZKContainer.NAME);
+            zkContainer.start();
+
+            createKeysAndTokens(zkContainer);
+        }
+
+        final String clusterName = Stream.of(this.getClass().getSimpleName(), randomName(5))
+                .filter(s -> s != null && !s.isEmpty())
+                .collect(joining("-"));
+
+        PulsarClusterSpec spec = PulsarClusterSpec.builder()
+                .numBookies(2)
+                .numBrokers(1)
+                .clusterName(clusterName)
+                .build();
+
+        log.info("Setting up cluster {} with token authentication  and {} bookies, {} brokers",
+                spec.clusterName(), spec.numBookies(), spec.numBrokers());
+
+        pulsarCluster = PulsarCluster.forSpec(spec);
+
+        for (BrokerContainer brokerContainer : pulsarCluster.getBrokers()) {
+            configureBroker(brokerContainer);
+            brokerContainer.withEnv("authenticationEnabled", "true");
+            brokerContainer.withEnv("authenticationProviders",
+                    "org.apache.pulsar.broker.authentication.AuthenticationToken");
+            brokerContainer.withEnv("authorizationEnabled", "true");
+            brokerContainer.withEnv("superUserRoles", "super-user");
+        }
+
+        pulsarCluster.start();
+
+        log.info("Cluster {} is setup", spec.clusterName());
+    }
+
+    @AfterSuite
+    @Override
+    public void tearDownCluster() {
+        super.tearDownCluster();
+    }
+
+    @Override
+    public String getTestName() {
+        return "token-auth-test-suite";
+    }
+
+    @Test
+    public void testPublishWithTokenAuth() throws Exception {
+        final String tenant = "token-test-tenant" + randomName(4);
+        final String namespace = tenant + "/ns-1";
+        final String topic = "persistent://" + namespace + "/topic-1";
+
+        @Cleanup
+        PulsarAdmin admin = PulsarAdmin.builder()
+                .serviceHttpUrl(pulsarCluster.getHttpServiceUrl())
+                .authentication(AuthenticationFactory.token(superUserAuthToken))
+                .build();
+
+        admin.tenants().createTenant(tenant,
+                new TenantInfo(Collections.singleton("regular-user"),
+                        Collections.singleton(pulsarCluster.getClusterName())));
+
+        admin.namespaces().createNamespace(namespace, Collections.singleton(pulsarCluster.getClusterName()));
+        admin.namespaces().grantPermissionOnNamespace(namespace, "regular-user", EnumSet.allOf(AuthAction.class));
+
+        @Cleanup
+        PulsarClient client = PulsarClient.builder()
+                .serviceUrl(pulsarCluster.getPlainTextServiceUrl())
```

Comment:
not for this PR, but do we have a requireTLS option? We should have an option to block authorization unless TLS is in use to avoid people sending tokens over cleartext.

Reply:
I think there was another PR that was meant to allow to disable the clear-text service endpoints.